mitmproxy/proxy/server.py

```diff
@@ -76,6 +88,8 @@ class ConnectionHandler(metaclass=abc.ABCMeta):
     client: Client
     max_conns: typing.DefaultDict[Address, asyncio.Semaphore]
     layer: layer.Layer
+    wakeuptimer: typing.Optional[WakeupTimer] = None
+    wakeuptimer_task: typing.Optional[asyncio.Task] = None
```

Comment:
We may have multiple timers here, which would currently overwrite each other. Let's make this a simple `wakeup_timer: set[asyncio.Task]`, I don't think we need to store the instances at all. :)

Reply:
OK, I will just cancel all elements in `wakeup_timer` on shutdown. The remembered `RequestWakeup` instances in the `command_sources[command]` in `http/__init__.py` do not require any cleanup, right?

Reply:
No, all of that is automatically garbage-collected. :)